plugins/GSdx/GSPng.cpp

```diff
@@ -20,15 +20,59 @@
 
 #include "stdafx.h"
 #include "GSPng.h"
+#include <png.h>
 
 namespace GSPng {
 
+    void SaveFile(const string& file, uint8** image, uint32 width,
+            uint32 height, int channel_bit_depth, int type, bool rb_swapped = false)
+    {
+        FILE *fp = fopen(file.c_str(), "wb");
+        if (fp == nullptr)
+            return;
+
+        png_structp png_ptr = png_create_write_struct(PNG_LIBPNG_VER_STRING,
+                nullptr, nullptr, nullptr);
+        if (png_ptr == nullptr) {
+            fclose(fp);
+            return;
+        }
+
+        png_infop info_ptr = png_create_info_struct(png_ptr);
+        if (info_ptr == nullptr) {
+            png_destroy_write_struct(&png_ptr, nullptr);
+            fclose(fp);
+            return;
+        }
+
+        if (setjmp(png_jmpbuf(png_ptr))) {
+            png_destroy_write_struct(&png_ptr, nullptr);
+            fclose(fp);
+            return;
+        }
+
```

Comment:
Because *this file* is *this file*, the *following file* is the *following file*? ;)

If I did that, the style in this file would be inconsistent. Also, I prefer my braces on the same line.

Well, I just noticed I made the style slightly inconsistent in the GSTexture{9,11}.cpp files...